server/controllers/feeds.ts

```diff
@@ -127,40 +133,81 @@ async function generateVideoFeed (req: express.Request, res: express.Response) {
   const account = res.locals.account
   const videoChannel = res.locals.videoChannel
   const nsfw = buildNSFWFilter(res, req.query.nsfw)
+  const format = req.query.format || req.params.format || 'rss'
 
-  const { name, description, imageUrl } = buildFeedMetadata({ videoChannel, account })
-
-  const feed = initFeed({
-    name,
-    description,
-    imageUrl,
-    resourceType: 'videos',
-    queryString: new URL(WEBSERVER.URL + req.url).search
-  })
+  const { name, description, imageUrl, accountImageUrl, email, link, accountLink } = await buildFeedMetadata({ videoChannel, account })
 
   const options = {
     accountId: account ? account.id : null,
     videoChannelId: videoChannel ? videoChannel.id : null
   }
 
+  console.log(`isLocal: ${req.query.isLocal}$`)
+
   const server = await getServerActor()
   const { data } = await VideoModel.listForApi({
     start,
     count: CONFIG.FEEDS.VIDEOS.COUNT,
-    sort: req.query.sort,
+    sort: format === 'podcast' ? '-publishedAt' : req.query.sort,
     displayOnlyForFollower: {
       actorId: server.id,
       orLocalVideos: true
     },
     nsfw,
-    isLocal: req.query.isLocal,
+    // Prevent podcast feeds from listing videos in other instances
+    // helps prevent duplicates when they are indexed -- only the author should control them
+    isLocal: format === 'podcast' ? true : req.query.isLocal,
     include: req.query.include | VideoInclude.FILES,
+    // TODO: include tags for future inclusion into the RSS feed
+    // include: req.query.include | VideoInclude.FILES | VideoInclude.TAGS,
     hasFiles: true,
     countVideos: false,
     ...options
   })
 
-  addVideosToFeed(feed, data)
+  let feed: Feed
+
+  if (format === 'podcast') {
```

Comment:
I think you should create a dedicated endpoint for podcast feed (like `/feeds/podcast/videos`). This way you would not accept `isLocal` or `sort` parameters.